### PR TITLE
Or 1397 titan canyon unify usdc.e usdc bridge into tokamak contracts bedrock src l2 tokamak

### PR DESCRIPTION
**Pull request by xxeonge**
1. Unifying USDC.e and L2UsdcBridge contracts.
2. Add submodules used by the contracts and remapped their paths in foundry.toml
3. Linked library for FiatTokenV2_2.sol to the predeploy address of SignatureCheck.sol

### CODE REVIEW

**Comment by boohyung** [comment]
@xxeonge Thanks a lot. But the Slither CI test is not passing properly because of some error. Did you check it?
```
Failed to generate IR for L2UsdcBridge.onlyEOA.
```

**Comment by AaronLee22** [comment]
> L2UsdcBridge.onlyEOA.



> @xxeonge Thanks a lot. But the Slither CI test is not passing properly because of some error. Did you check it?
> 
> ```
> Failed to generate IR for L2UsdcBridge.onlyEOA.
> ```

Let me check this out too!

**Comment by nguyenzung** [comment]
Correct me if I am wrong. I think we can extend OptimismMintableERC20 and StandardBridge for this task.

Actually, I don't know about the background. What is the benefit of this solution?

**Comment by xxeonge** [comment]
> @xxeonge Thanks a lot. But the Slither CI test is not passing properly because of some error. Did you check it?
> 
> ```
> Failed to generate IR for L2UsdcBridge.onlyEOA.
> ```

Sorry for late reply. I will check. Thank you!!

**Comment by xxeonge** [comment]
++) The L1UsdcBridge contract was added to predeploy the L1Usdcbridge.

 4. remove libraries tag in foundry.toml, I'll remove it first, as it's better suited for predeploy not unifying.
 5. Unifying L1UsdcBridge and proxy contracts.
 6. libraries and universal related to UsdcBridge have been integrated into the canyon repo.
 7. Modified slither-check.yml to resolve slither errors.

File: packages/tokamak/contracts-bedrock/foundry.toml [hunk 1]
```diff
@@ -24,6 +26,7 @@ bytecode_hash = 'none'
 build_info = true
 build_info_path = 'artifacts/build-info'
 
+
```
Comment (rlgns98kr) [inline packages/tokamak/contracts-bedrock/foundry.toml:29]: Thank you!
I think we can remove this empty line!

**Comment by rlgns98kr** [comment]
@AaronLee22 
Thank you very much!
Do you know why the check action is not worked about latest commit?

File: packages/tokamak/contracts-bedrock/foundry.toml (hunk 1 above)
Comment (xxeonge) [reply packages/tokamak/contracts-bedrock/foundry.toml]: @rlgns98kr 
Thank you! i removed empty lines : )

**Comment by AaronLee22** [comment]
> @AaronLee22 Thank you very much! Do you know why the check action is not worked about latest commit?


Sure, I attempted to resolve this error by following both recommendations from the Slither team, specifically by removing the 'ignore-compile: true' option, yet the issue persisted. The Slither team is currently addressing a known problem where certain libraries or aliases within smart contracts cannot be retrieved. Consequently, I was compelled to manually exclude the specific smart contract experiencing the 'Failed to generate IR' issue.
Screenshot 2024-04-02 at 10 27 57 AM
Screenshot 2024-03-27 at 10 28 05 AM

File: packages/tokamak/contracts-bedrock/src/libraries/Address.sol [hunk 2]
```diff
@@ -0,0 +1,222 @@
+// SPDX-License-Identifier: MIT
```
Comment (rlgns98kr) [inline packages/tokamak/contracts-bedrock/src/libraries/Address.sol:1]: I'm sorry. I missed these files.
How about changing the directory structure? I'm sorry again my fault,,, 😢😢

src
  tokamak-contracts
    USDC
      L1
      L2
      libraries
      universal

Comment (xxeonge) [reply packages/tokamak/contracts-bedrock/src/libraries/Address.sol:1]: @rlgns98kr 
In L2, is there no need to separate the usdc contract and usdcBridge contract files??

Comment (rlgns98kr) [reply packages/tokamak/contracts-bedrock/src/libraries/Address.sol:1]: Oh, I'm sorry for my late.
About sub directory, I think yours is greeat! So I think you can keep your seperate structure in the L2!

**Review by rlgns98kr — APPROVED** [review]
Thank you very much, @xxeonge and @AaronLee22!
Congratulations for your first contribution! 👍👍

File: packages/tokamak/contracts-bedrock/src/tokamak-contracts/USDC/L2/tokamak-USDC/upgradeability/UpgradeabilityProxy.sol [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+     */
+    event Upgraded(address implementation);
+
+    /**
+     * @dev Storage slot with the address of the current implementation.
+     * This is the keccak-256 hash of "org.zeppelinos.proxy.implementation", and is
+     * validated in the constructor.
+     */
+    bytes32
+        private constant IMPLEMENTATION_SLOT = 0x7050c9e0f4ca769c69bd3a8ef740bc37934f8e2c036e5a723fd8ee048ed3f8c3;
```
Comment (nguyenzung) [inline packages/tokamak/contracts-bedrock/src/tokamak-contracts/USDC/L2/tokamak-USDC/upgradeability/UpgradeabilityProxy.sol:46]: @xxeonge It is the impl slot for usdc

File: packages/tokamak/contracts-bedrock/src/tokamak-contracts/USDC/L2/tokamak-USDC/upgradeability/AdminUpgradeabilityProxy.sol [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+     */
+    event AdminChanged(address previousAdmin, address newAdmin);
+
+    /**
+     * @dev Storage slot with the admin of the contract.
+     * This is the keccak-256 hash of "org.zeppelinos.proxy.admin", and is
+     * validated in the constructor.
+     */
+    bytes32
+        private constant ADMIN_SLOT = 0x10d6a54a4754c8869d6886b5f5d7fbfa5b4522237ea5c60d11bc4e7a1ff9390b;
```
Comment (nguyenzung) [inline packages/tokamak/contracts-bedrock/src/tokamak-contracts/USDC/L2/tokamak-USDC/upgradeability/AdminUpgradeabilityProxy.sol:45]: It is admin slot for usdc

**Pull request merged**